### PR TITLE
ref(images-loaded): Check 'Not Found' filter option by default if Image status equals 'Missing'

### PR DESCRIPTION
**Pull request by priscilawebdev**
. Add new filter logic: 
   - If the Image Status equals Missing, then check also 'Not Found' by default
   - If one of the filter's category has one option, then the filter won't display this category
 
. Fix a minor issue with DropDownMenu
   - Change the prop children to a function or an element was necessary, so then the dropdown menu can be customized
   
**Before:**
image

**After:**
image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/dropdownControl.tsx [hunk 1]
```diff
@@ -77,35 +92,40 @@ class DropdownControl extends React.Component<Props> {
...
-      blendWithActor,
-      priority,
-      className,
-    } = this.props;
+  renderChildren(isOpen: boolean, getMenuProps: GetMenuPropsFn) {
+    const {children, alignRight, menuWidth, blendWithActor, priority} = this.props;
+
+    if (typeof children === 'function') {
+      return children({isOpen, getMenuProps});
+    }
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/dropdownControl.tsx:100]: We have utility for this:
https://github.com/getsentry/sentry/blob/488044c4573d9403976b5ec0978af7a3609593d9/src/sentry/static/sentry/app/utils/callIfFunction.tsx#L1-L4

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/filter/index.tsx [hunk 2]
```diff
@@ -60,28 +58,38 @@ function Filter({options, onFilter, className}: Props) {
...
-            </React.Fragment>
-          ))}
-        </Content>
+        {({getMenuProps, isOpen}) => (
+          <StyledContent
+            {...getMenuProps()}
+            alignMenu="left"
+            width="240px"
+            isOpen={isOpen}
+            className="drop-down-filter-menu"
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/filter/index.tsx:67]: Is this className used?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/filter/index.tsx:67]: yes 😉

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/filter/index.tsx:67]: here https://github.com/getsentry/sentry/blob/218e6e036092b2f4cdb3708820c5e743df4c2617/src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx#L394

File: src/sentry/static/sentry/app/components/dropdownControl.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dropdownControl.tsx:100]: Yes I know this function, but I also remember Evan saying something about not using it, so I didn't use it

File: src/sentry/static/sentry/app/components/dropdownControl.tsx [hunk 3]
```diff
@@ -20,7 +20,21 @@ type DefaultProps = {
...
+type ButtonArgs = {
+  isOpen: boolean;
+  getActorProps: GetActorPropsFn;
+};
+
 type Props = DefaultProps & {
+  children:
+    | ((args: ChildrenArgs) => React.ReactElement)
+    | React.ReactElement
+    | Array<React.ReactNode>;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/dropdownControl.tsx:37]: Could this be `React.ReactNode | React.ReactNode[]` or do you only want a single element and not text?

File: src/sentry/static/sentry/app/components/dropdownControl.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/dropdownControl.tsx:100]: I don't think callIfFunction is a good fit here as you still need to use typeof to know if you can return.

File: src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx [hunk 4]
```diff
@@ -134,37 +140,52 @@ class Candidates extends React.Component<Props, State> {
...
+    const {imageStatus} = this.props;
+
+    const filterOptions = {};
+
+    const candidateStatus = [
+      ...new Set(candidates.map(candidate => candidate.download.status)),
+    ];
+
+    if (candidateStatus.length > 1) {
+      filterOptions[filterOptionCategories.status] = candidateStatus.map(status => ({
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx:152]: Are the keys to `filterOptions` also translated strings?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx:152]: yes they are

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx:152]: yes https://github.com/getsentry/sentry/blob/5bb78cf4421b8259deb99efb14e0c3eb0178b8ce/src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx#L24

File: src/sentry/static/sentry/app/components/dropdownControl.tsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dropdownControl.tsx:37]: As `React.ReactNode` contains functions, I am getting a type conflict. That is why I can't use it

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/dropdownControl.tsx]: I updated the types:

```
children:
    | ((args: ChildrenArgs) => React.ReactElement)
    | React.ReactElement
    | Array<React.ReactElement>;
```

**Review by markstory — APPROVED** [review]

**Pull request merged**